wasmtime/_instance.py

```diff
@@ -49,40 +50,59 @@ def __from_ptr__(cls, ptr, module):
             raise TypeError("wrong pointer type")
         ty.__ptr__ = ptr
         ty._module = module
+        ty._exports = None
         return ty
 
     @property
     def exports(self):
         """
         Returns the exports of this module
-        """
-        externs = ExternTypeList()
-        dll.wasm_instance_exports(self.__ptr__, byref(externs.vec))
-        ret = []
-        for i in range(0, externs.vec.size):
-            ret.append(wrap_extern(externs.vec.data[i], externs))
-        return ret
 
-    def get_export(self, name):
-        """
-        Gets an export from this module by name, returning `None` if the name
-        doesn't exist.
+        The returned type can be indexed both with integers and with strings for
+        names of exports.
         """
-        if not hasattr(self, '_export_map'):
-            self._export_map = {}
-            exports = self.exports
-            for i, export in enumerate(self._module.exports):
-                self._export_map[export.name] = exports[i]
-        if name in self._export_map:
-            return self._export_map[name]
-        else:
-            return None
+        if self._exports is None:
+            externs = ExternTypeList()
+            dll.wasm_instance_exports(self.__ptr__, byref(externs.vec))
+            extern_list = []
+            for i in range(0, externs.vec.size):
+                extern_list.append(wrap_extern(externs.vec.data[i], externs))
+            self._exports = InstanceExports(extern_list, self._module)
+        return self._exports
 
     def __del__(self):
         if hasattr(self, '__ptr__'):
             dll.wasm_instance_delete(self.__ptr__)
 
 
+class InstanceExports(object):
+    def __init__(self, extern_list, module):
+        self.extern_list = extern_list
+        self.extern_map = {}
```

Comment:
I'd probably make these members `_private_ just in case the implementation changes in the future (exposing them doesn't provide any new capability in the API AFAICT).